src/storage/src/healthcheck.rs

```diff
@@ -820,6 +833,48 @@ mod tests {
         );
     }
 
+    #[mz_ore::test]
+    #[cfg_attr(miri, ignore)] // Reports undefined behavior
+    fn test_health_operator_write_namespaced_map() {
+        use Step::*;
+
+        // Test 2 inputs across 2 workers.
+        health_operator_runner(
+            2,
+            2,
+            // testing this
+            false,
+            vec![
+                AssertStatus(vec![
+                    // Assert both inputs started.
+                    StatusToAssert {
+                        collection_index: 0,
+                        status: "starting".to_string(),
+                        ..Default::default()
+                    },
+                    StatusToAssert {
+                        collection_index: 1,
+                        status: "starting".to_string(),
+                        ..Default::default()
+                    },
+                ]),
+                Update(TestUpdate {
+                    worker_id: 0,
+                    namespace: StatusNamespace::TestScript,
+                    input_index: 1,
+                    update: HealthStatusUpdate::stalled("uhoh".to_string(), None),
+                }),
+                AssertStatus(vec![StatusToAssert {
+                    collection_index: 1,
+                    status: "stalled".to_string(),
+                    error: Some("testscript: uhoh".to_string()),
+                    errors: None,
```

Comment:
Just to be clear: it's only whether to write in `details` but it will still write the `error` with namespace

Reply:
yeah, this is what `errors` captures!